lib/targets/cuda/malloc.cpp

```diff
@@ -573,6 +685,13 @@ namespace quda
         }
         pool_init = true;
       }
+#if defined(NVSHMEM_COMMS)
+      MPI_Comm tmp = MPI_COMM_WORLD;
```

Comment:
One thing to be mindful of---the default MPI space may not be `MPI_COMM_WORLD`, and that doesn't necessarily mean split grid is being used. This is exposed in the QUDA interface here: https://github.com/lattice/quda/blob/develop/lib/interface_quda.cpp#L361

I'm not sure if there's a great solution to this, since a user passing in their own custom handle _could_ mean they're managing split-grid on their own, and I don't know how that'll play with `nvshmem`... ugh.

Reply:
This gets leveraged in MILC (via the MILC interface exposure) here: https://github.com/milc-qcd/milc_qcd/blob/develop/generic/milc_to_quda_utilities.c#L35 ; tracing the routine `mycomm` becomes a bit of a rabbit hole.

Reply:
We can pass in MPI_COMM_HANDLE_USER here if this has been set. Or we can disallow setting a different handle here for now with nvshmem. We need to do some more investigation with split_grid anyway.

Reply:
For the time being we can just add a warning around using NVSHMEM with split grid, either within QUDA or as done by other apps. Something along the lines of QUDA's NVSHMEM work assumes your MPI world == MPI_COMM_WORLD?